src/journal/journald-server.c

```diff
@@ -2591,8 +2600,172 @@ static void server_load_credentials(Server *s) {
         }
 }
 
-int server_new(Server **ret) {
+static void server_set_defaults(Server *s) {
+        assert(s);
+
+        s->storage = s->namespace ? STORAGE_PERSISTENT : STORAGE_AUTO;
+
+        s->compress.enabled = true;
+        s->compress.threshold_bytes = UINT64_MAX;
+
+        s->seal = true;
+
+        s->read_kmsg = !s->namespace;
+
+        s->set_audit = true;
+        s->sync_interval_usec = DEFAULT_SYNC_INTERVAL_USEC;
+
+        s->ratelimit_interval = DEFAULT_RATE_LIMIT_INTERVAL;
+        s->ratelimit_burst = DEFAULT_RATE_LIMIT_BURST;
+
+        s->system_storage.name = "System Journal";
+        journal_reset_metrics(&s->system_storage.metrics);
+
+        s->runtime_storage.name = "Runtime Journal";
+        journal_reset_metrics(&s->runtime_storage.metrics);
+
+        s->max_retention_usec = 0;
+        s->max_file_usec = DEFAULT_MAX_FILE_USEC;
+
+        s->forward_to_syslog = false;
+        s->forward_to_kmsg = false;
+        s->forward_to_console = false;
+        s->forward_to_wall = true;
+        s->forward_to_socket = (SocketAddress) { .sockaddr.sa.sa_family = AF_UNSPEC };
+
+        s->tty_path = NULL;
+
+        s->max_level_store = LOG_DEBUG;
+        s->max_level_syslog = LOG_DEBUG;
+        s->max_level_kmsg = LOG_NOTICE;
+        s->max_level_console = LOG_INFO;
+        s->max_level_wall = LOG_EMERG;
+        s->max_level_socket = LOG_DEBUG;
+
+        s->split_mode = SPLIT_UID;
+        s->line_max = DEFAULT_LINE_MAX;
+}
+
+static void server_set_config(Server *s) {
+        int r;
+
+        assert(s);
+
+        server_set_defaults(s);
+        server_load_credentials(s);
+        server_parse_config_file(s);
```

Comment:
On reload, it is not necessary to load credentials. Please keep configs somewhere and reuse them on reload.